drivers/HwWatchdog.h

```diff
@@ -0,0 +1,118 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_H
+#define MBED_WATCHDOG_H
+
+#ifdef DEVICE_WATCHDOG
+
+#include "watchdog_api.h"
+
+#include "mbed_error.h"
+
+#include "rtos/Mutex.h"
+
+#include "mbed.h"
+
+#include "LowPowerTimeout.h"
+
+#include <cstdio>
+
+namespace mbed {
+/** \addtogroup drivers */
+/** A system timer that will reset the system in the case of system failures or
+ *  malfunctions.
+ *
+ * Example:
+ * @code
+ *
+ * HwWatchdog watchdog;
+ * watchdog.start();
```

Comment:
example shows just `start` API for watchdog, should we be describing other uses cases as well in example?

Reply:
once we added the sw watchdog and interface between HW watchdog to SW watchdog ,later we will update/add more description